topics/bloomfilters.md

```diff
@@ -58,6 +58,84 @@ In this username example, we can use a Bloom filter to track every username that
 * If no, the user is created and the username is added to the Bloom filter.
 * If yes, the app can decide to either check the main database or reject the username.
 
+## Tutorial
+
+**There are two ways to run valkey-bloom:**
+   - **Docker**: Follow the steps here for using the [valkey bundle](valkey-bundle.md)
+   - **Build from source**: Follow the [build instructions](https://github.com/valkey-io/valkey-bloom/blob/unstable/README.md#build-instructions)
+
+
+### Usage example with CLI
+
+* Create a bloom filter of taken usernames with a 1 in 1000 chance of false positives and 10000 initial capacity.
+```bash
+127.0.0.1:6379> BF.RESERVE usernames 0.001 10000
+OK
+```
+* Add usernames which now have signed up to the bloom filters
+```bash
+# Add one user to the bloom filter
+127.0.0.1:6379> BF.ADD usernames johnsmith
+(integer) 1
+
+# Add multiple users at once
+127.0.0.1:6379> BF.MADD usernames JaneDoe valkeyFan bloomEnjoyer
+1) (integer) 1
+2) (integer) 1
+3) (integer) 1
+```
+
+* Check if usernames have been taken already
+```bash
+# Check if as single user has been added
+127.0.0.1:6379> BF.EXISTS usernames johnsmith
+(integer) 1
+127.0.0.1:6379> BF.EXISTS usernames fake_user
+(integer) 0
+
+# Check if multiple users are present in the bloomfilter at once
+127.0.0.1:6379> BF.MEXISTS usernames johnsmith fake_user JaneDoe
+1) (integer) 1
+2) (integer) 0
+3) (integer) 1
+```
+
+* Check how many users have been added to the filter
+```bash
+127.0.0.1:6379> BF.CARD usernames
+(integer) 4
+```
+* View your bloom filter object's usage:
+```bash
+127.0.0.1:6379> BF.INFO usernames
+ 1) Capacity
+ 2) (integer) 10000
+ 3) Size
+ 4) (integer) 23912
+ 5) Number of filters
+ 6) (integer) 1
+ 7) Number of items inserted
+ 8) (integer) 4
+ 9) Error rate
+10) "0.001"
+11) Expansion rate
+12) (integer) 2
+13) Tightening ratio
+14) "0.5"
+15) Max scaled capacity
+16) (integer) 26214300
+```
+* If you know the exact number of users who will sign up, or if want to limit the number of items regardless, you can create a non scaling bloom filter. This will create a filter that has capacity for only 1000 items with a 1 in 10,000 chance of a false positive.
```

Comment:
Do you mean 1 in 100,000?

The example below uses 0.00001. We can correct either one